project-fob/Controllers/AttendeeController.cs

```diff
@@ -27,24 +27,37 @@ public void Fob(string value)
             var gotvalue = HttpContext.Session.TryGetValue("meetingid", out var meetingIdValue);
 
             string byteArrayToString = System.Text.Encoding.ASCII.GetString(meetingIdValue);
-            Fob fob = db.Fob.Include(x => x.Meeting).ThenInclude(x => x.Stats).Include(x => x.fobbed).SingleOrDefault(f => f.Meeting.MeetingId == byteArrayToString);
+            Fob fob = db.Fob.Include(x => x.Meeting).ThenInclude(x => x.Stats).Include(x => x.fobbed).ThenInclude(x => x.User).SingleOrDefault(f => f.Meeting.MeetingId == byteArrayToString);
             if (fob == null)
             {
                 throw new ArgumentNullException();
-            } 
+            }
             // if fob.fobbed doesnt contain the current attendee
 
             gotvalue = HttpContext.Session.TryGetValue("sessionid", out var session);
 
             string byteArrayToString2 = System.Text.Encoding.ASCII.GetString(session);
-            Attendee att = db.Attendee.SingleOrDefault(at => at.User.UserId.Equals((byteArrayToString2)));
+            Attendee att = db.Attendee.Include(at => at.User).Include(at => at.Meeting).SingleOrDefault(at => at.User.UserId.Equals(byteArrayToString2) && at.Meeting.MeetingId.Equals(byteArrayToString));
 
-            if (att == null) 
+            if (att == null)
             {
                 throw new ArgumentNullException();
             }
 
-            if (!fob.fobbed.Contains(att))
+
+            //Need a rework as if someone is fobbing it it'll not get if the person is already fobbed because of the different primary key for different session.
+            //A workaround would be to replace attendee with user as we don't need the attendee anymore as the meeting (or fob) contains the meeting id anyways.
+            bool foundfobber = false;
+
+            for (int i = 0; i < fob.fobbed.Count; i++)
+            {
+                if (fob.fobbed[i].Equals(att))
+                {
+                    foundfobber = true;
+                    i = fob.fobbed.Count;
```

Comment:
Is `i` being changed to just to stop the loop? If so, you can use a `break` instead. It's pretty unusual to change the `for` loop variable inside the loop.

Given that this seems to just be checking the `fob.fobbed` list contains `att`, you could replace the loop with something like `bool foundfobber = fob.fobbed.Any(x => x.Equals(att));` - but if you want to keep the loop for some reason, then it might be nice to extract it to another private method that just returns true/false if `att` is in the list.

Reply:
Yeah majority of the time I'm not using break because I've been taught that using break is a bad solution and it should not be used. (It's more of a bad habit of mine from academic years because I already looked it up a few times and it's not that bad nowadays. (And I was a bit lazy to implement a do while loop)